### PR TITLE
[13.0] Migrate queue, add pre-commit with black, pylint, flake8, ...

### PR DESCRIPTION
**Pull request by guewen**
* Migrate queue_job, queue_job_cron, queue_job_subscribe to 13.0.
* Add the new OCA's pre-commit configuration

### CODE REVIEW

**Comment by etobella** [comment]
Lint fails due to E501, it should be removed from LINT for v13 on maintainer-tools

**Comment by bealdav** [comment]
Some `super(MyClass, self).` instead of `super().` but not blocking for me.

And black à part ;-)

Impressive the job of black

**Comment by guewen** [comment]
> Lint fails due to E501, it should be removed from LINT for v13 on maintainer-tools

Thanks, you are right. However, even the local pre-commit fails with W503 errors although they are in the ignored list in `.flake.cfg`.

https://github.com/OCA/queue/pull/175/files#diff-540ed609f086a01ee6f84e1a7bba5f6eR10

**Comment by guewen** [comment]
> However, even the local pre-commit fails with W503 errors although they are in the ignored list in `.flake.cfg`.

Figured that out, there is a second pre-commit check only for `__init__.py` files to ignore unused imports, so the list of ignore is overidden by the command line option in `.pre-commit-config.yaml`. Pushed a fixup.

**Comment by sebalix** [comment]
@guewen not installable modules should be excluded from the checks. @jcoux has done something for that internally, an `exclude:` statement is generated in the `pre-commit` config file, and once the module is migrated you have to remove it from this list manually (another script has to be run on Travis to compare current modules with the pre-commit exclude list).

**Comment by sbidoul** [comment]
More config improvements and evolution in https://github.com/guewen/queue/pull/7

**Comment by guewen** [comment]
> More config improvements and evolution in [guewen#7](https://github.com/guewen/queue/pull/7)

Thanks, it's very clean now. And everything is green :champagne:

**Comment by sbidoul** [comment]
@guewen even more improvements in https://github.com/guewen/queue/pull/8

I think I'm done now :)

**Comment by guewen** [comment]
Rebased

**Comment by sbidoul** [comment]
@guewen before merging wait a little bit for the linter configs to stabilize, then squash everything related to that in one commit.

File: .travis.yml
Comment (yvaucher) [inline .travis.yml]: @guewen should we push this change on cache into https://github.com/OCA/maintainer-quality-tools/pull/613 ?

Comment (guewen) [reply .travis.yml]: @yvaucher this was from an older configuration I guess.

**Comment by guewen** [comment]
Rebased from 13.0 where @sbidoul pushed the new pre-commit/travis configuration.
The last commit contains the updates from black and isort.

File: base_export_async/__manifest__.py
Comment (Tonow-c2c) [inline base_export_async/__manifest__.py:9]:
```suggestion
    "version": "13.0.1.0.0",
```

File: base_import_async/__manifest__.py
Comment (Tonow-c2c) [inline base_import_async/__manifest__.py:8]:
```suggestion
    "version": "13.0.1.0.0",
```

File: queue_job/readme/HISTORY.rst
Comment (yvaucher) [inline queue_job/readme/HISTORY.rst:14]: I'm curious, do we really want to remove previous version history?

**Review by yvaucher — APPROVED** [review]
LGTM

Didn't reviewed the black, isort commit

Non blocking question about changelog

File: base_import_async/models/base_import_import.py
Comment (Tonow-c2c) [inline base_import_async/models/base_import_import.py:39]:
```suggestion
            return super().do(fields, options, dryrun=dryrun)
```

File: base_import_async/models/base_import_import.py
Comment (Tonow-c2c) [inline base_import_async/models/base_import_import.py:35]: :fire:

File: base_import_async/models/queue_job.py
Comment (Tonow-c2c) [inline base_import_async/models/queue_job.py:12]: :fire:

File: test_base_import_async/__manifest__.py
Comment (Tonow-c2c) [inline test_base_import_async/__manifest__.py:7]:
```suggestion
    "version": "13.0.1.0.0",
```

File: queue_job/models/queue_job.py
Comment (Tonow-c2c) [inline queue_job/models/queue_job.py:172]:
```suggestion
        res = super().write(vals)
```

File: queue_job/models/queue_job.py
Comment (Tonow-c2c) [inline queue_job/models/queue_job.py:338]:
```suggestion
        return super().unlink()
```

File: base_import_async/models/base_import_import.py
Comment (Tonow-c2c) [inline base_import_async/models/base_import_import.py:81]: :fire:

File: base_export_async/__manifest__.py
Comment (guewen) [reply base_export_async/__manifest__.py:9]: this module is not migrated yet

File: queue_job/readme/HISTORY.rst
Comment (guewen) [reply queue_job/readme/HISTORY.rst:14]: Not sure, but as anyway the history is very partial, I'm not sure it's useful. As the branching point is indicated, it's maybe better to look in the 12.0 branch if we want it.

**Comment by Tonow-c2c** [comment]
Sorry @guewen I made a mistake like `Black` changed files of modules that are not migrated I thought everything was being migrated :disappointed:

**Review by Tonow-c2c — APPROVED** [review]

**Review by simahawk — APPROVED** [review]
skimmed through it. LG

**Pull request merged**
